conans/cli/commands/list.py

```
@@ -104,6 +119,27 @@ def json_formatter(info):
 }
 
 
+def _add_remotes_and_cache_options(subparser):
+    remotes_group = subparser.add_mutually_exclusive_group()
+    remotes_group.add_argument("-r", "--remote", default=None, action=Extender,
+                               help="Name of the remote to add")
+    remotes_group.add_argument("-a", "--all-remotes", action='store_true',
+                               help="Search in all the remotes")
+    subparser.add_argument("-c", "--cache", action='store_true', help="Search in the local cache")
+
+
+def _get_remotes(conan_api, args):
+    remotes = []
+    error = None
+    if args.all_remotes:
+        remotes, error = conan_api.get_active_remotes(None)
+    elif args.remote:
+        remotes, error = conan_api.get_active_remotes(args.remote)
+    if error:
+        raise ConanException(error)
```

If the API method returns an error, why delegate the exception to the caller? Let the API method raise the exception instead of returning a tuple